src/db/couchDevices.ts

```diff
@@ -63,10 +74,36 @@ const loginIdDesign = makeJsDesign('loginId', ({ emit }) => ({
   }
 }))
 
+/**
+ * Looks up devices by the IP location sorted by city.
+ */
+const locationByCity = makeJsDesign('locationByCity', ({ emit }) => ({
+  map: function (doc) {
+    if (doc.location == null) return
+    const { country = '', region = '', city = '' } = doc.location
```

Comment:
I think this will crash in the production environment - CouchDB runs a hideously old JS engine that doesn't even have `const`, let alone destructuring. I would propose:

```js
emit([
  doc.location.country || '',
  doc.location.region || '',
  doc.location.city || ''
}, null)
```

And then tell ESLint to leave us alone on this one.

Reply:
It appeared to have worked in my local version of couch, but better to be safe than sorry I suppose.